pkg/controller/infrastructure/infraflow/ensurer.go

```diff
@@ -755,11 +755,24 @@ func (fctx *FlowContext) DeleteResourceGroup(ctx context.Context) error {
 
 // DeleteSubnetsInForeignGroup deletes all managed subnets in a foreign resource group
 func (fctx *FlowContext) DeleteSubnetsInForeignGroup(ctx context.Context) error {
-	vnetCfg := fctx.adapter.VirtualNetworkConfig()
-	if vnetCfg.Managed {
+	if fctx.adapter.VirtualNetworkConfig().Managed {
 		return nil
 	}
 
+	// This is a prerequisite for the deletion of the subnets in foreign resource group because
+	// internal load balancers might have a Frontend IP configuration referencing the
+	// foreign subnet which therefore can not be deleted. Since the Frontend IP configuration
+	// by its own can not be deleted, we remove the whole (all) load balancers.
+	err := fctx.deleteLoadBalancers(ctx)
+	if err != nil {
+		return err
+	}
```

Comment:
Overall, looks great. I would probably move this over to it's own "node" in the graph rather than doing it with the subnets.

lgtm otherwise